### DISCUSSION TITLE
Add feature to perform firmware updates

### DISCUSSION THREAD

**Discussion by nick-leroy** [hmatuschek/qdmr#291] [Ideas]
qDMR has been a huge quality of life improvement for many of us, however when we need to do firmware updates on our radios we're still having to go back to the manufacturers' software to perform this action.

**Comment by hmatuschek** [comment]
I understand that it would be nice to have this feature, but there are some issues with firmware updates, that prevent me from implementing it. First of all, a firmware update is very device specific and usually uses a different protocol than the one used to program the codeplug. As qdmr deals with the codeplug and not the firmare,  I would not implement it. There are, however, several open firmware projects for several devices. These projects usually provide a cross-platform tool to update the firmware.

Secondly, programming the codeplug is a relatively save operation. In the worst case, you got a non-functional configuration of your device. If something goes wrong during the firmware update, the device might be bricked.
